tools/test/toolchains/test_toolchains.py

```diff
@@ -70,18 +127,88 @@ def test_gcc_minimal_printf(self):
         for i in minimal_printf_wraps:
             self.assertIn(i, gcc_obj.flags["ld"])
 
+    def test_gcc_arm_default_lib(self):
+        """Test that linker flags are correctly added to an instance of GCC_ARM."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.supported_c_libs = {"gcc_arm": ["small"]}
+        mock_target.default_lib = "sMALL"
+        mock_target.supported_toolchains = ["GCC_ARM"]
+        mock_target.is_TrustZone_secure_target = False
+        gcc_arm_obj = GCC_ARM(mock_target)
+        self.assertIn("-DMBED_RTOS_SINGLE_THREAD", gcc_arm_obj.flags["common"])
+        self.assertIn("-D__NEWLIB_NANO", gcc_arm_obj.flags["common"])
+        self.assertIn("--specs=nano.specs", gcc_arm_obj.flags["ld"])
+
+    def test_gcc_arm_default_lib_std_exception(self):
+        """Test that an exception raised if the default_lib is std but it is not listed in supported_c_libs."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.default_toolchain = "ARM"
+        mock_target.default_lib = "std"
+        mock_target.supported_c_libs = {"arm": ["small"]}
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+            GCC_ARM(mock_target)
+
+    def test_gcc_arm_default_lib_small_exception(self):
+        """Test that an exception raised if the default_lib is small but it is not listed in supported_c_libs."""
+        mock_target = mock.MagicMock()
+        mock_target.core = "Cortex-M4"
+        mock_target.default_lib = "small"
+        mock_target.supported_c_libs = {"arm": ["std"]}
+        mock_target.default_toolchain = "ARM"
+        mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+            GCC_ARM(mock_target)
 
 class TestIarToolchain(TestCase):
     """Test the IAR class."""
 
     def test_iar_minimal_printf(self):
-        """Test that linker flags are correctly added to an instance of GCC_ARM."""
+        """Test that linker flags are correctly added to an instance of IAR."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
         mock_target.printf_lib = "minimal-printf"
         mock_target.supported_toolchains = ["IAR"]
+        mock_target.default_lib = "std"
+        mock_target.supported_c_libs = {"iar": ["std"]}
         mock_target.is_TrustZone_secure_target = False
 
         iar_obj = IAR(mock_target)
         var = "-DMBED_MINIMAL_PRINTF"
         self.assertIn("-DMBED_MINIMAL_PRINTF", iar_obj.flags["common"])
+
+    def test_iar_default_lib(self):
+        """Test that case insensitive of the default_lib option works without exception."""
```

Comment:
The test case is testing that no exception is raised when a valid c library is specified.

Reply:
I have not added the std lib support in IAR but added changes to work for case insensitive so this test case testing case insensitive.

Reply:
What you have added is to check if a valid c library is specified.  You are literally calling `self.check_c_lib_supported(target, "iar")` in `iar.py` which raises an exception if an unsupported c library type is specified. Therefore as per my original comment
>The test case is testing that no exception is raised when a valid c library is specified.